src/components/Link.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+  <router-link v-if="this.options.link" :to="this.options.link">
+    <slot />
+  </router-link>
+  <a class="clickable" v-else @click="this.click" >
+    <slot />
+  </a>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop } from 'vue-property-decorator';
+
+export interface LinkOptions {
```

Comment:
Again, `options` somewhat confuses me. For me, the name implies, that it renders or includes a list of several links, the user can choose from. But it is more a `LinkElement` or simple `Link` where the programmer has the choice to either provide a link or a onClick method.